proto/opamp.proto

```diff
@@ -277,6 +277,16 @@ message OpAMPConnectionSettings {
     // If this field has no value or is set to 0, the Agent should not send any heartbeats.
     // Status: [Development]
     uint64 heartbeat_interval_seconds = 4;
+
+    // Additional connection settings. These are Agent-specific and are up to the Agent
```

Comment:
Mark all new fields and messages with `Status: [Development]`. New features start at this level.

Reply:
I [posted](https://github.com/open-telemetry/opamp-spec/issues/203#issuecomment-2399956499) earlier:
>One other design consideration is to make sure OpAMPConnectionSettings, TelemetryConnectionSettings and OtherConnectionSettings are defined uniformly, with the exception of settings which are only applicable to a particular connection type (e.g. heartbeat_interval_seconds is unique to OpAMPConnectionSettings). A possible option would be to move all common settings to one proto message and include that from OpAMPConnectionSettings, TelemetryConnectionSettings and OtherConnectionSettings. That would be a breaking change, so we will need to discuss if it is worth it.

@andykellr what do you think?

Reply:
I would definitely like to avoid breaking changes at this point. I think the approach of having additional messages for TLS and Proxy settings creates enough uniformity between the connection settings messages.

I likely won't be able to review this PR until early next week, but I will take a look. At first glance it looks reasonable but you raise good questions.

Reply:
OK, let's avoid the breaking changes.